shared/spanner.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2018 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package shared
+
+//
+// Shared data types used for string WPT test results in Cloud Spanner.
+//
+
+const (
+	// TestStatusUnknown is an uninitialized TestStatus and should
+	// not be used.
+	TestStatusUnknown int64 = 0
+
+	// TestStatusPass indicates that all tests completed successfully and passed.
+	TestStatusPass int64 = 1
+
+	// TestStatusOK indicates that all tests completed successfully.
+	TestStatusOK int64 = 2
+
+	// TestStatusError indicates that some tests did not complete
+	// successfully.
+	TestStatusError int64 = 3
+
+	// TestStatusTimeout indicates that some tests timed out.
+	TestStatusTimeout int64 = 4
+
+	// TestStatusNotRun indicates that a test was not run.
+	TestStatusNotRun int64 = 5
+
+	// TestStatusFail indicates that a test failed.
+	TestStatusFail int64 = 6
+
+	// TestStatusCrash indicates that the WPT test runner crashed attempting to run the test.
+	TestStatusCrash int64 = 7
+
+	// TestStatusDefault is the default value used when a status string cannot be
+	// interpreted.
+	TestStatusDefault int64 = 0
+
+	// TestStatusNameDefault is the default string used when a status value cannot
+	// be interpreted.
+	TestStatusNameDefault string = "UNKNOWN"
+)
+
+var testStatusValues = map[string]int64{
```

Comment:
I can't think of a use case for converting from string to int. If we don't have one right now, let's remove this map, the `TestStatusNameDefault` constant and the related function. We can always add them back later.

Reply:
The use case is parsing WPT reports for storing data in Cloud Spanner.